compat/endian.h

```diff
@@ -44,6 +44,9 @@
 #endif
 #endif /* __linux__ */
 
+#if defined(__FreeBSD__) && !defined(HAVE_ENDIAN_H)
+#include <sys/endian.h>
+#endif
```

Comment:
I think we will still need the `#define betoh16 be16toh` part in this case.
So probably it should read:

```
#if defined(__FreeBSD__)
#if !defined(HAVE_ENDIAN_H)
#include <sys/endian.h>
#endif
#if !defined(betoh16)
#define betoh16	be16toh
...
```

Reply:
Yes, That is the lines following this statement that make the extra definitions

but your patch with the nested `defined`s works too